code/backend/batch/utilities/tools/question_answer_tool.py

```diff
@@ -145,8 +169,39 @@ def answer_question(self, question: str, chat_history: list[dict], **kwargs):
 
         llm_helper = LLMHelper()
 
-        response = llm_helper.get_chat_completion(messages, temperature=0)
+        response = llm_helper.get_chat_completion(messages, model=model, temperature=0)
+        clean_answer = self.format_answer_from_response(
+            response, question, source_documents
+        )
+
+        return clean_answer
+
+    def create_image_url_list(self, source_documents):
+        image_types = self.config.get_advanced_image_processing_image_types()
+
+        image_documents = []
+
+        for doc in source_documents:
```

Comment:
Apologies, I'm reading this in the browser. But it seems we iterate around one list and then iterate around it again to add the sas token - could we do this all in one iteration?

Reply:
Good point, done